File: Source/Java/datalabs/string/PartialFormatter.java
package datalabs.string;

import java.util.Map;
import java.util.regex.Matcher;
import java.util.regex.Pattern;

public class PartialFormatter {
    public static String format(String value, Map<String, Object> parameters) {
        String resolvedValue = value;

        for (Map.Entry<String, Object> entry : parameters.entrySet()) {
            String key = entry.getKey();

            resolvedValue = resolvedValue.replaceAll("\\{" + entry.getKey() + "\\}", entry.getValue().toString());
        }

        return resolvedValue;
    }
}

File: Source/Kotlin/datalabs/string/PartialFormatter.kt
package datalabs.string

import java.util.*

object PartialFormatter {
    @JvmStatic
    fun format(value: String, parameters: Map<String, Any>): String {
        var resolvedValue = value

        for ((key, value) in parameters) {
            resolvedValue = resolvedValue.replace("\\{$key\\}".toRegex(), value.toString())
        }

        return resolvedValue
    }
}
